src/Integration.UnitTests/Suppression/ChecksumCalculatorTests.cs

```diff
@@ -0,0 +1,105 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarLint.VisualStudio.Integration.Suppression;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.Suppression
+{
+    [TestClass]
+    public class ChecksumCalculatorTests
+    {
+        [TestMethod]
+        public void Checksum_ThrowsOnNull()
+        {
+            Exceptions.Expect<ArgumentNullException>(() => ChecksumCalculator.Calculate(null));
+        }
+
+        [TestMethod]
+        public void Checksum_WhitespaceIsIgnored()
+        {
+            // 1. If the input only varies by whitespace then the checksums should be the same
+            var checksum1 = ChecksumCalculator.Calculate("abc");
+            var checksum2 = ChecksumCalculator.Calculate(" a  b  \r\n\t c \n\n");
+            Assert.AreEqual(checksum1, checksum2);
+
+            // 2. Logically, a whitespace-only string should have the same checksum as an empty tring
+            var emptyChecksum = ChecksumCalculator.Calculate("");
+            var whitespaceOnlyChecksum = ChecksumCalculator.Calculate("\r \t\n\n\r\t  ");
+            Assert.AreEqual(emptyChecksum, whitespaceOnlyChecksum);
+        }
+
+        [TestMethod]
+        public void Checksum_ExpectedValues()
+        {
+            // Alpha-numerical
+            AssertExpectedChecksum("1234567890abcdefghijklmnopqrstuvwxyz", "928f7bcdcd08869cc44c1bf24e7abec6");
```

Comment:
The tricky part of writing the tests was knowing what the expected value should be.
For this test method, I wrote a small Java program to encode the test strings using the same code as SLI/SLE.
For the next method, I called the SQ batch/issues API to real issues for Akka.Net to get the checksum values.

Reply:
What about leaving this comment in the code in case we make some change later on and need to change the values?

Reply:
Done

Reply:
Done